src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,258 @@
+    @Optional() private phoneFieldComponent: SkyPhoneFieldComponent
+  ) { }
+
+  public ngOnInit(): void {
+    if (!this.phoneFieldComponent) {
+      throw new Error(
+        'You must wrap the `skyPhoneFieldInput` directive within a ' +
+        '`<sky-phone-field>` component!'
+      );
+    }
+
+    const element = this.elRef.nativeElement;
+
+    this.adapterService.addElementClass(element, 'sky-form-control');
+    this.adapterService.setElementPlaceholder(element,
+      this.phoneFieldComponent.selectedCountry.placeholder);
+
+    this.adapterService.setAriaLabel(element);
+  }
+
+  public ngAfterViewInit(): void {
+    this.phoneFieldComponent.selectedCountryChange
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((country: SkyCountryData) => {
+        // Write the value again to cause validation to refire
+        this.writeValue(this.modelValue);
+        this.adapterService.setElementPlaceholder(this.elRef.nativeElement, country.placeholder);
+      });
+
+    // This is needed to address a bug in Angular 4, where the value is not changed on the view.
+    // See: https://github.com/angular/angular/issues/13792
+    /* istanbul ignore else */
+    if (this.control && this.modelValue) {
+      this.control.setValue(this.modelValue, { emitEvent: false });
+        this.changeDetector.detectChanges();
+    }
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  /**
+   * Writes the new value for reactive forms on change events on the input element
+   * @param event The change event that was received
+   */
+  @HostListener('change', ['$event'])
+  public onInputChange(event: any): void {
+    this.writeValue(event.target.value);
+  }
+
+  /**
+   * Marks reactive form controls as touched on input blur events
+   */
+  @HostListener('blur')
+  public onInputBlur(): void {
+    this.onTouched();
+  }
+
+  /**
+   * Writes the new value for reactive forms
+   * @param value The new value for the input
+   */
+  public writeValue(value: any): void {
+    this.modelValue = value;
+  }
+
+  public registerOnChange(fn: (value: any) => any): void { this.onChange = fn; }
+
+  public registerOnTouched(fn: () => any): void { this.onTouched = fn; }
+
+  public registerOnValidatorChange(fn: () => void): void { this.validatorChange = fn; }
+
+  /**
+   * Sets the disabled state on the input
+   * @param isDisabled the new value of the input's disabled state
+   */
+  public setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  /**
+   * Validate's the form control's current value
+   * @param control the form control for the input
+   */
+  public validate(control: AbstractControl): ValidationErrors {
+
+    if (!this.control) {
+      this.control = control;
+    }
+
+    let value = control.value;
+
+    if (!value) {
+      return;
+    }
+
+    if (!this.validateNumber(value) && !this.skyPhoneFieldNoValidate) {
```

Comment:
We could throw a "short-circuit" return statement at the top of the `validate()` method to make sure that any future validations we add are skipped when `this.skyPhoneFieldNoValidate` is set to `true`.

(Also, another developer wouldn't need to parse out why `this.skyPhoneFieldNoValidate` was paired with `this.validateNumber()` in the same condition.)

```
if (!this.control) {
  this.control = control;
}

if (this.skyPhoneFieldNoValidate) {
  return;
}

const value = control.value;

if (!value) {
  return;
}

if (!this.validateNumber(value)) { }

if (yetAnotherValidationFails) { }

// etc.
```

Reply:
Good call. Done